test/Microsoft.ML.Benchmarks/TrainPredictionBench.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+using Microsoft.ML.Models;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Transforms;
+
+namespace Microsoft.ML.Benchmarks
+{
+    public class TrainPredictionBench
+    {
+        internal static ClassificationMetrics s_metrics;
```

Comment:
>nternal static ClassificationMetrics s_metrics; [](start = 9, length = 47)

Would also be awesome to figure out how to minimize the amount of code per dataset.  so that you can just add a dataset and then not have to change much else
Also, it should be good to be able to quickly and easily substitute different learners.  
Basically the dimensions for tests are:
Training:
  Task:
    Learner
       Hyper-parameters{ detault, sweep}
    Dataset
       Subset by size (like 10%, 50%, 100% of rows)

Reply:
Yeah, this is a good idea, but I think such generalizations are best done when actually trying to develop the second, third, etc. test. i.e. I will try to do it when I am adding more tests.